samples/cpp/prompt_lookup_decoding_lm/prompt_lookup_decoding_lm.cpp

```diff
@@ -47,13 +47,18 @@ struct TextStreamer {
         if (text.size() >= 3 && text.compare(text.size() - 3, 3, "�") == 0) {
             // Don't print incomplete text
             return;
+        } else if (text.size() > print_len && std::any_of(text.begin() + print_len, text.end(), [](char c) { return c == ' '; })) {
+            // It is possible to have a shorter text after adding new token.
+            // Print to output only if text lengh is increaesed and if containt full words.
+            std::cout << std::string_view{text.data() + print_len, text.size() - print_len} << std::flush;
+            print_len = text.size();
         }
-        std::cout << std::string_view{text.data() + print_len, text.size() - print_len} << std::flush;
-        print_len = text.size();
     }
 
     void end() {
         std::string text = detokenize(detokenizer, token_cache);
+        if (text.size() < print_len)
+            return ;
```

Comment:
1. The inversion of `text.size() > print_len` from `put()` is `text.size() <= print_len`, not `text.size() < print_len`.
2. Why isn't space required here?

Reply:
1. thanks for noticing, fixed
2. spaces are not strictly required, removed them from above